### PR TITLE
[SPARK-10036][SQL]Load JDBC driver in DataFrameReader.jdbc and DataFrameWriter.jdbc

### PR DESCRIPTION
**Pull request by zsxwing**
This PR uses `JDBCRDD.getConnector` to load JDBC driver before creating connection in `DataFrameReader.jdbc` and `DataFrameWriter.jdbc`.

### CODE REVIEW

**Comment by zsxwing** [comment]
DataFrameReader.jdbc and DataFrameWriter.jdbc ignore `extraOptions` currently. Is it intentional, or a bug?

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 1]
```diff
@@ -124,6 +124,8 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
    * Construct a [[DataFrame]] representing the database table accessible via JDBC URL
    * url named table and connection properties.
    *
+   * Note: Options in this reader will be ignored.
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: i think that's a bug

Comment (zsxwing) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: Okey. Fixed it.

**Comment by marmbrus** [comment]
Thanks!  Merged to master and 1.5

**Pull request closed without merging**